File: 02Algorithm_Study/04_week_homeWork/phone_number.py
def solution(phoneBook):
    phoneBook = sorted(phoneBook)

    for p1, p2 in zip(phoneBook, phoneBook[1:]):
        print("p1",p1,"p2",p2)
        if p2.startswith(p1):
            return False
    return True

#  B.startswith(A) 내가 찾고자 하는 문자열 A가 문자열 B 

solution(["119", "97674223", "1195524421"])

File: 02Algorithm_Study/phone_number.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;

    for(string s1 : phone_book){
        for(string s2 : phone_book){
            if(s1 == s2){
                continue;
            }
            if(s2.length() > s1.length()){
                if(s1 == s2.substr(0,s1.length())){
                    return false;
                }
            }
        }
    }

    return answer;
}